pkg/ui/src/Toasts/__tests__/toaster.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { test, describe, expect, afterEach } from "vitest";
+import { cleanup, render } from "@testing-library/svelte";
+import Fragment from "svelte-fragment-component";
+import html from "svelte-htm";
+
+import { get } from "svelte/store";
+
+import { createToasterStore, createToaster } from "../toaster";
+
+import TestToaster from "./TestToaster.svelte";
+
+const defaultToast = {
+	message: "I'm a toast",
+	duration: 2000
+};
+
+afterEach(() => {
+	cleanup();
+});
+
+describe("toasterStore", () => {
+	test("should push new toasts to queue and assign them an id", () => {
+		const toaster = createToasterStore();
+
+		toaster.push(defaultToast);
+
+		const [toast] = get(toaster);
+
+		expect(toast).toHaveProperty("id");
+	});
+
+	test("should pop toasts by id", () => {
+		const toaster = createToasterStore();
+
+		toaster.push(defaultToast);
+
+		const [toast] = get(toaster);
+
+		toaster.pop(toast.id);
+
+		const toasts = get(toaster);
+
+		expect(toasts.length).toBe(0);
+	});
```

Comment:
I would provide some 'noise' here: add three toasts and remove the middle one (to assert it doesn't take out last/first, but the one with a given id